### PR TITLE
Add secure-vault changes.

### PR DESCRIPTION
**Pull request by hasinidilanka**
### Proposed changes in this pull request

- Change the `org.wso2.securevault` version to 1.1.2

-  Resolve the secrets using`MiscellaneousUtil.resolve()` method.

- Depends on: [https://github.com/wso2/carbon-secvault/pull/64](https://github.com/wso2/carbon-secvault/pull/64)

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java [hunk 1]
```diff
@@ -225,28 +225,35 @@ protected static String getPropertyValue(String key) {
...
-            SecretResolver secretResolver;
-
-            if (StringUtils.isBlank(secretProvider)) {
-                properties.put(SECRET_PROVIDER, DEFAULT_CALLBACK_HANDLER);
+        String secretProvider = (String) properties.get(SECRET_PROVIDER);
+        if (StringUtils.isBlank(secretProvider)) {
+            properties.put(SECRET_PROVIDER, DEFAULT_CALLBACK_HANDLER);
+        }
+        SecretResolver secretResolver = SecretResolverFactory.create(properties);
+        Enumeration propertyNames = properties.propertyNames();
```
Comment (pulasthi7) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: Use map entry set iterator instead

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConfigBuilder.java [hunk 2]
```diff
@@ -229,28 +230,23 @@ private void resolveSecrets() {
...
-                    // Resolving the secret password.
-                    String value = secretResolver.resolve(key);
-                    // Replaces the original encrypted property with resolved property
-                    notificationMgtConfigProperties.put(key, value);
-                } else {
-                    if (log.isDebugEnabled()) {
-                        log.debug("No encryption done for value with key :" + key);
-                    }
+                String value = notificationMgtConfigProperties.getProperty(key);
+                if (value != null){
```
Comment (pulasthi7) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConfigBuilder.java]: formatting

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConfigBuilder.java [hunk 3]
```diff
@@ -229,28 +230,23 @@ private void resolveSecrets() {
...
+                    value = MiscellaneousUtil.resolve(value, secretResolver);
+
                 }
+                notificationMgtConfigProperties.put(key, value);
             }
-        } else {
+        }
+        else {
```
Comment (pulasthi7) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConfigBuilder.java]: formatting

File: components/notification-mgt/org.wso2.carbon.identity.notification.mgt/src/main/java/org/wso2/carbon/identity/notification/mgt/NotificationMgtConfigBuilder.java [hunk 4]
```diff
@@ -238,26 +239,19 @@ private void resolveSecrets() {
 
         SecretResolver secretResolver = SecretResolverFactory.create(notificationMgtConfigProperties);
         Enumeration propertyNames = notificationMgtConfigProperties.propertyNames();
```
Comment (pulasthi7) [inline components/notification-mgt/org.wso2.carbon.identity.notification.mgt/src/main/java/org/wso2/carbon/identity/notification/mgt/NotificationMgtConfigBuilder.java]: any usages of this?

File: components/carbon-authenticators/thrift-authenticator/org.wso2.carbon.identity.authenticator.thrift/src/main/java/org/wso2/carbon/identity/thrift/authentication/internal/util/ThriftAuthenticationConfigParser.java [hunk 5]
```diff
@@ -137,11 +138,16 @@ private void readChildElements(OMElement serverConfig, Stack<String> nameStack)
...
                 String key = getKey(nameStack);
                 Object currentObject = configuration.get(key);
-                String value = replaceSystemProperty(element.getText());
-                if (secretResolver != null && secretResolver.isInitialized() &&
-                        secretResolver.isTokenProtected(key)) {
-                    value = secretResolver.resolve(key);
+                String value;
+                String resolvedValue = MiscellaneousUtil.resolve(element, secretResolver);
+
+                if (resolvedValue != null && !resolvedValue.isEmpty()) {
```
Comment (denuwanthi) [inline components/carbon-authenticators/thrift-authenticator/org.wso2.carbon.identity.authenticator.thrift/src/main/java/org/wso2/carbon/identity/thrift/authentication/internal/util/ThriftAuthenticationConfigParser.java]: Use StringUtils isNotEmpty

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java [hunk 6]
```diff
@@ -107,10 +109,8 @@ public static synchronized void init() {
                     //Initialize the keystores in EndpointConfig.properties only if the "mutualSSLManagerEnabled"
                     // feature is enabled.
                     if (isMutualSSLManagerEnabled(getPropertyValue(Constants.TenantConstants.MUTUAL_SSL_MANAGER_ENABLED))) {
-                        if (isSecuredPropertyAvailable(prop)) {
-                            // Resolve encrypted properties with secure vault
-                            resolveSecrets(prop);
-                        }
+                        // Resolve encrypted properties with secure vault
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: Issue with comment format

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java [hunk 7]
```diff
@@ -225,28 +226,35 @@ protected static String getPropertyValue(String key) {
...
+                String value = entry.getValue().toString();
+                if (value != null) {
+                    value = MiscellaneousUtil.resolve(value, secretResolver);
+                }
+                properties.put(key, value);
             }
-
-            secretResolver = SecretResolverFactory.create(properties, "");
+        }
+        // Support the protectedToken alias used for encryption. ProtectedToken alias is deprecated
```
Comment (JKAUSHALYA) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: Fix comment formatting issues in all instances.

File: components/carbon-authenticators/thrift-authenticator/org.wso2.carbon.identity.authenticator.thrift/src/main/java/org/wso2/carbon/identity/thrift/authentication/internal/util/ThriftAuthenticationConfigParser.java (hunk 5 above)
Comment (hasinidilanka) [reply components/carbon-authenticators/thrift-authenticator/org.wso2.carbon.identity.authenticator.thrift/src/main/java/org/wso2/carbon/identity/thrift/authentication/internal/util/ThriftAuthenticationConfigParser.java]: Fixed - https://github.com/wso2/carbon-identity-framework/pull/2621

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java (hunk 6 above)
Comment (hasinidilanka) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: Fixed - https://github.com/wso2/carbon-identity-framework/pull/2621

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java (hunk 7 above)
Comment (hasinidilanka) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint.util/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/util/MutualSSLManager.java]: Fixed - https://github.com/wso2/carbon-identity-framework/pull/2621

**Pull request merged**